p2panda-rs/src/operation/traits.rs

```diff
@@ -61,7 +65,30 @@ pub trait AsOperation {
 /// [`StorageProvider`][crate::storage_provider::traits::StorageProvider] implementations should
 /// implement this for a data structure that represents an operation as it is stored in the
 /// database.
-pub trait AsVerifiedOperation {
+pub trait AsVerifiedOperation:
+    Sized + Clone + Send + Sync + Validate + PartialEq + std::fmt::Debug
+{
+    /// Error type for `AsVerifiedOperation`
+    type VerifiedOperationError: 'static + std::error::Error + Send + Sync;
+
+    /// Returns a new `VerifiedOperation` instance.
+    ///
+    /// Use `VerifiedOperation::new_from_entry()` instead if you want to validate that the operation
```

Comment:
Hmm, lolz, yeh this is weird. Is it just a naming thing? If this was called `AuthoredOperation` then it would be less weird? 

Maybe the method names could be reversed? `new()` takes entry and operation and verifies. Maybe we can place the other behind a testing flag actually, I don't think it is hugely useful in the wild. These methods aren't actually new in this PR though. So maybe I should open an issue and deal with this in a separate PR.

Reply:
I raised an issue for this: https://github.com/p2panda/p2panda/issues/366

Reply:
I'm gunna deal with this in a follow-up PR as the issue wasn't introduced in this PR.